sklearn/utils/_param_validation.py

```diff
@@ -154,6 +155,10 @@ def validate_params(parameter_constraints):
         Note that the *args and **kwargs parameters are not validated and must not be
         present in the parameter_constraints dictionary.
 
+    prefer_skip_nested_validation : bool, default=False
+        If True, the validation of parameters of inner estimators or functions
+        called by the decorated function will be skipped.
```

Comment:
Could you please explain why this is useful to set it to `True` for specific cases (calling public function from another public function that has already validated all the public facing parameter) and give more detailed guide line on when to use it or not use it in scikit-learn?

And maybe also contrast it with the use of the global skip_parameter_validation` config param.

Reply:
Basically reusing https://github.com/scikit-learn/scikit-learn/pull/25815#discussion_r1172712342.